### PR TITLE
Port k2 unit tests to skv

### PR DESCRIPTION
**Pull request by ahsank**

### CODE REVIEW

File: src/SKVClient/test/SKVRecordTest.cpp [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+    w.write(storage);
+    Binary buf;
+    REQUIRE(w.flush(buf));
+    MPackReader reader(buf);
+    dto::SKVRecord::Storage read_storage{};
+    REQUIRE(reader.read(read_storage));
+    dto::SKVRecord reconstructed("collection", std::make_shared<dto::Schema>(schema),
+                                     std::move(read_storage), true);
+
+    std::optional<String> last = reconstructed.deserializeNext<String>();
```
Comment (ahsank) [inline src/SKVClient/test/SKVRecordTest.cpp:255]: @iafuture Everything works except this line causes double free error. The stack trace doesn't have symbols. Any suggestion how to debug this issue?

File: src/SKVClient/src/skv/dto/SKVRecord.cpp [hunk 2]
```diff
@@ -266,11 +268,23 @@ SKVRecord SKVRecord::getSKVKeyRecord() {
...
+    }
+
+    // Exclude all value fields
+    for (size_t i = num_keys; i < excludedFields.size(); ++i) {
+        excludedFields[i] = true;
+    }
 
-    for (size_t i = 0; i < schema->partitionKeyFields.size() + schema->rangeKeyFields.size(); ++i) {
-        if (!storage.excludedFields[i]) {
+    for (size_t i = 0; i < excludedFields.size(); ++i) {
```
Comment (iafuture) [inline src/SKVClient/src/skv/dto/SKVRecord.cpp:286]: this should only iterate up to num_keys, not the full number of fields

Comment (ahsank) [reply src/SKVClient/src/skv/dto/SKVRecord.cpp:286]: Non key fields needs to have null values.  If it only iterates up to num_keys there won't be any value for non key fields.

Comment (iafuture) [reply src/SKVClient/src/skv/dto/SKVRecord.cpp:286]: That's taken care of by the resize call above - it initializes the entire array to false

Comment (ahsank) [reply src/SKVClient/src/skv/dto/SKVRecord.cpp:286]: But it's not affecting the builder. The new record is created from builder. One has to populate all fields in builder.

**Review by iafuture — APPROVED** [review]

**Pull request merged**